app/src/main/java/io/github/sds100/keymapper/util/ui/ResourceProvider.kt

```diff
@@ -65,4 +69,5 @@ interface ResourceProvider {
     fun getText(@StringRes resId: Int): CharSequence
     fun getDrawable(@DrawableRes resId: Int): Drawable
     fun getColor(@ColorRes color: Int): Int
+    fun getContext(): Context;
```

Comment:
This is dangerous! We don't want to be exposing Context in viewmodels. It is a code-smell and can also lead to memory leaks. The SpinnerAdapter should be in the Fragment since it is only needed for the View.

Reply:
Sorry about that. I knew you will mention it, but as a lag of knowhow in Android Dev this was my only solution. I changed it and removed the getContext function